### PR TITLE
Issue #302: modify EmptyStatement to use checkNamesForContextStrategyByTokenOrParentSet property

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #302: add EmptyStatement and EqualsHashCode to CHECK_TO_ANCESTOR_NODES_MAP

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java [hunk 1]
```diff
@@ -53,6 +53,10 @@
     static {
         CHECK_TO_ANCESTOR_NODES_MAP.put("AvoidNestedBlocks",
                 Arrays.asList(TokenTypes.SLIST));
+        CHECK_TO_ANCESTOR_NODES_MAP.put("EmptyStatement",
+                Arrays.asList(TokenTypes.METHOD_DEF));
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:57]: METHOD_DEF is too big scope and doesn't cover statements in constructors, lambdas, static blocks, etc. [Example AST](http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=72fba57f86eaf7df54f418b381c28422&code=93063b278d27defddea640338e439801&checkstyle=checkstyle-8.35-all.jar&printTree=true&printSuppressions=).

Why can't we use parent node and `checkNamesForContextStrategyByTokenOrParent` property?

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java [hunk 2]
```diff
@@ -53,6 +53,10 @@
...
+        CHECK_TO_ANCESTOR_NODES_MAP.put("EqualsHashCode",
+                Arrays.asList(TokenTypes.CLASS_DEF));
```
Comment (baratali) [inline src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:59]: I agree with EqualsHashCode. Please move it to separate pull request and I can merge it to master.

**Review by baratali — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 1 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:57]: @baratali `checkNamesForContextStrategyByTokenOrParent` is ok, sorry, maybe it's my oversight.

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 2 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:57]: ok, separate PR https://github.com/checkstyle/patch-filters/pull/325

Comment (baratali) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java:57]: thanks. please remove this EqualsHashCode from here

**Review by baratali — CHANGES_REQUESTED** [review]
please rebase

File: src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java (hunk 2 above)
Comment (HuGanghui) [reply src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java]: ok, removed.

**Comment by HuGanghui** [comment]
@baratali rebased

**Review by baratali — APPROVED** [review]

**Pull request merged**